resources/datadog-integrations/project.rb.erb

```diff
@@ -43,8 +43,11 @@ end
 dependency 'preparation'  # creates required build directories
 dependency 'dd-check-<%= name %>-software'
 
+# Checks should require the datadog agent
+runtime_dependency 'datadog-agent'
+
 extends_packages('datadog-agent', 'datadog')
 
-extra_package_file '/etc/dd-agent/conf.d/<%= name %>.yaml.example'
+extra_package_file '/etc/dd-agent/conf.d/examples/<%= name %>.yaml.example'
```

Comment:
You're right, defaults should be shipped with the agent, not with the checks themselves.
The only case I can see is if we publish a new package for a default check with a config change (because I think all packages will have their packages in the end, even if they are also shipped with the agent). But then I guess one user wanting to upgrade the check before the release of the agent will know the config changes to make. 
So I'd say that we don't have to ship defaults, only examples, and nothing to change.